model/model.py

```diff
@@ -184,45 +189,46 @@ def validate_batch(self):
 
     # TODO funktionen gör alldeles för mycket,
     # dela upp utskrift, beräkning och träning
-    def train(self):
+    def train(self, use_pretrained_net=False):
         """ Trains the model on the dataset """
         print("Starting training...")
 
-        if self.use_pretrained:
+        if self.use_pretrained and \
+                (self.use_pretrained_net and use_pretrained_net) or \
+                (not self.use_pretrained_net and not use_pretrained_net):
             self._session.run(self.embedding_init,
                               feed_dict={self.embedding_placeholder:
                                          self.data.embedding_matrix})
-        self.train_writer = \
```

Comment:
Perhaps not necessary but otherwise the writer initiated twice, once for each time we train.